src/main.rs

```diff
@@ -559,6 +566,47 @@ fn run() -> Result<()> {
                 )
             })?;
 
+            if app_matches.is_present("list-languages") {
+                let languages = assets.syntax_set.syntaxes();
+
+                let longest = languages.iter()
+                    .map(|s| s.name.len())
+                    .max()
+                    .unwrap_or(32); // Fallback width if they have no language definitions.
+
+                let separator = " | ";
+                for lang in languages {
+                    if lang.hidden {
+                        continue;
+                    }
+                    print!("{:width$}{}", lang.name, separator, width = longest);
+
+                    // Line-wrapping for the possible file extension overflow.
+                    let desired_width = options.term_width - longest - separator.len();
+                    // Number of characters on this line so far, wrap before `desired_width`
+                    let mut num_chars = 0;
+
+                    let comma_separator = ", ";
+                    let mut extension = lang.file_extensions.iter().peekable();
+                    while let Some(word) = extension.next() {
+                        // If we can't fit this word in, then create a line break and align it in.
+                        if word.len() + num_chars + comma_separator.len() >= desired_width {
+                            num_chars = 0;
+                            print!("\n{:width$}{}", "", separator, width = longest);
+                        }
+
+                        num_chars += word.len();
```

Comment:
Actually, this also needs to be `num_chars += word.len() + comma_separator.len();`. But I can also change this, if you want.